### PR TITLE
Alerting: Disable /api/admin/pause-all-alerts with Unified Alerting

### LINKED ISSUE #51729: Alerting: `/api/admin/pause-all-alerts` should not have an effect when Legacy Alerting is disabled.
`/api/admin/pause-all-alerts` makes a direct call to the SQLStore to mark every alert in the legacy alerting system as "paused." We've had reports of confusion when unified alerting is enabled, because this API will still return a message like: `{"alertsAffected":1144,"message":"alerts paused","state":"paused"}`, but then seem to have no effect. That's because it's pausing legacy alerts that are not being evaluated at all.

If the API is called when unified alerting is enabled, it should return a status code and message that makes it clear that it has had no effect.

### PR DESCRIPTION
**Pull request by joeblubaugh**
/api/admin/pause-all-alerts only takes effect for legacy alerts. This
change returns a 403 if it's called when unified alerting is enabled.

Fixes #51729

### CODE REVIEW

**Review by papagian — COMMENTED** [review]
LGTM but I have a question

File: pkg/api/alerting.go [hunk 1]
```diff
@@ -507,91 +507,107 @@ func (hs *HTTPServer) NotificationTest(c *models.ReqContext) response.Response {
...
-		return response.Error(http.StatusBadRequest, "alertId is invalid", err)
-	}
-	result := make(map[string]interface{})
-	result["alertId"] = alertID
-
-	query := models.GetAlertByIdQuery{Id: alertID}
-	if err := hs.SQLStore.GetAlertById(c.Req.Context(), &query); err != nil {
-		return response.Error(500, "Get Alert failed", err)
+func (hs *HTTPServer) PauseAlert(ngalertEnabled bool) func(c *models.ReqContext) response.Response {
+	if ngalertEnabled {
```
[the remark this answers is not in the document]
Comment (joeblubaugh) [reply pkg/api/alerting.go]: I've updated this to use `setting.AlertingEnabled`.

File: pkg/api/alerting.go [hunk 2]
```diff
@@ -507,91 +507,107 @@ func (hs *HTTPServer) NotificationTest(c *models.ReqContext) response.Response {
...
-		return response.Error(http.StatusBadRequest, "alertId is invalid", err)
-	}
-	result := make(map[string]interface{})
-	result["alertId"] = alertID
-
-	query := models.GetAlertByIdQuery{Id: alertID}
-	if err := hs.SQLStore.GetAlertById(c.Req.Context(), &query); err != nil {
-		return response.Error(500, "Get Alert failed", err)
+func (hs *HTTPServer) PauseAlert(legacyAlertingEnabled *bool) func(c *models.ReqContext) response.Response {
+	if legacyAlertingEnabled == nil || !*legacyAlertingEnabled {
```
Comment (papagian) [inline pkg/api/alerting.go:511]: Based on the following:
https://github.com/grafana/grafana/blob/62b0a8bae6abcbd75326d3310abebcf6a78eac78/pkg/setting/setting.go#L1427-L1431

https://github.com/grafana/grafana/blob/62b0a8bae6abcbd75326d3310abebcf6a78eac78/pkg/setting/setting_unified_alerting.go#L119-L122

I think that the only case `AlertingEnabled` is `nil` is when the `alerting.enabled` cannot be parsed and in that case the new alerting is enabled.
So, I think the check should become:
```suggestion
	if legacyAlertingEnabled != nil || !*legacyAlertingEnabled {
```

**Comment by yuri-tceretian** [comment]
Although I am OK with this approach, I wonder whether it would be better to just not register the endpoint at all, i.e. update code here:
https://github.com/grafana/grafana/blob/1d2aa7c69bf7db8cd4a5b7514c83ab17966f1790/pkg/api/api.go#L461-L467

**Review by papagian — CHANGES_REQUESTED** [review]

**Comment by joeblubaugh** [comment]
@yuri-tceretian I prefer the more informative message in this case - we haven't deprecated the API in the past or provided any other advanced notice.

**Comment by papagian** [comment]
> Although I am OK with this approach, I wonder whether it would be better to just not register the endpoint at all

I also like this suggestion unless we want to give better feedback to the users.
In any case I would suggest also to update [the docs](https://grafana.com/docs/grafana/latest/developers/http_api/admin/#pause-all-alerts) appropriately.

**Review by papagian — APPROVED** [review]

**Pull request merged**